cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -407,13 +408,78 @@ static void sc_mkdir_and_mount_and_bind(const char *rootfs_dir,
 	}
 }
 
+static int sc_mount_nvidia_is_driver_in_dir(const char *dir)
+{
+	char driver_path[512] = { 0 };
+
+	struct sc_nvidia_driver driver;
+
+	// Probe sysfs to get the version of the driver that is currently inserted.
+	sc_probe_nvidia_driver(&driver);
+
+	// If there's no driver then we should not bother ourselves with finding the
+	// matching library
+	if (driver.major_version == 0) {
+		return 0;
+	}
+	// Probe if a well known library is found in directory dir
+	sc_must_snprintf(driver_path, sizeof driver_path,
+			 "%s/" LIBNVIDIA_GLCORE_SO_PATTERN, dir,
```

Comment:
The word pattern may be confused with a glob pattern, maybe (nitpicky, I know this landed in master already) we could call it template.

Reply:
I'll put it in my TODO list and fix it once we merge the release branch back to master